great_expectations/experimental/datasources/file_path_data_asset.py

```diff
@@ -225,34 +227,39 @@ def get_batch_list_from_batch_request(
 
         return batch_list
 
+    def _build_or_get_data_connector(self) -> DataConnector:
+        """This private method ensures that exactly one instance of "DataConnector" class is available."""
+        data_connector: DataConnector
```

Comment:
I very-much dislike conditionally-created attributes because it makes the classes attributes less discoverable (esp: with type hinting) and forces callers to consider this `AttributeError`. self._data_connector should be initialized to empty or `None` in the constructor